### DISCUSSION TITLE
proposal for improvement file manager - Computers in LAN

### DISCUSSION THREAD

**Discussion by Mile-Lile** [linuxdeepin/developer-center#3131] [Features Request & Ideas | 特性请求 & 头脑风暴]
Hi deepin comunity,

here are some of my thoughts that bothers me about greatest linux distro... In File Manager, feature "Computers in LAN" needs improvement... it is misleading since I tried deepin for the first time, 3 years ago... It should either say "samba share" or should be improved with avahi-utils... because all people that used MS windows thinks of this option as a SD (service discovery) not as a SAMBA share... e.g. I just want to find shared printer in local netowork... for now it is very complicated... so maybe avahi should be instaled by default... or [wsdd2](https://packages.debian.org/unstable/net/wsdd2)

dde-dock:
Installed: 5.5.12-1
Candidate: 5.5.12-1
Version table:
*** 5.5.12-1 500
500 https://community-packages.deepin.com/deepin apricot/main amd64 Packages
100 /usr/lib/dpkg-db/status

dde-file-manager:
Installed: 5.5.10-1
Candidate: 5.5.10-1
Version table:
*** 5.5.10-1 500
500 https://community-packages.deepin.com/deepin apricot/main amd64 Packages
100 /us
![Screenshot_dde-desktop_20220425124151](https://user-images.githubusercontent.com/12752816/165078451-a27272d0-846c-4bfb-8c34-6f78ac162fa1.png)
r/lib/dpkg-db/status

**Comment by Mile-Lile** [comment]
if I don't install avahi or wsdd2 I usually get message "mounting error" which is misleading

**Comment by BLumia** [comment]
Since it's not a bug report, I'll move it to the discussion board :)

If I remember it correctly, the Computers in LAN section actually show all available results under [`network:///`](https://wiki.gnome.org/Projects/gvfs/schemes) witch merges results from `smb://` and `dns-sd://`. As you mentioned, packages like `avahi` should be installed to ensure `dns-sd://` works. It can be good to get that package pre-installed by default from the distro deepin. (cc @yxsos )

We cannot ensure if other distro that also provide DDE will also pre-install that package, but installing such package will make `dns-sd://` works, so I guess it shouldn't just called "samba share" ;)

**Comment by Mile-Lile** [comment]
@BLumia Thak you for reply! it is not bug but I didn't managed to post it on deepin forum (had to solve some puzzles and still couldn't post it, maybe admin blocked me or else), and on file-manager section (github) there are only pull requests...but there are some odds... I will post screenshots... If I look at log viewer I see that "Compuuter in LAN" uses smb:/// but that gives error when I try from CLI... on terminal smb:\\\ works... So maybe it is some trivial typo... smb:/// or smb:\\\ I have firewall enabled so please let someone test and comment here to see if works... Thx.... BTW network:/// works when I manually type in file-manager
![Screenshot_dde-desktop_20220425164444](https://user-images.githubusercontent.com/12752816/165115759-b79b57dc-066c-4534-9e03-dd63a5ff7c15.png)




![Screenshot_dde-desktop_20220425164650](https://user-images.githubusercontent.com/12752816/165115712-14a601ee-4005-491e-a930-23731ce09da4.png)

**Reply by BLumia** [reply]
> but I didn't managed to post it on deepin forum (had to solve some puzzles and still couldn't post it, maybe admin blocked me or else)

I'm not sure what "puzzle" you're referring to. If you can provide some details (p.s. I suggest creating a new discussion thread for this topic), maybe @myml could help check it out. And also, please feel free to use this discussion board as a forum too :)

> file-manager section (github) there are only pull requests

I have re-enabled the issue board for the file manager project, but it will still guide users to report issues to the developer-center repo. developer-center is the right place to report so we don't need to look into all the repos under the linuxdeepin organization.

> So maybe it is some trivial typo... smb:/// or smb:\\ I have firewall enabled so please let someone test and comment here to see if works... Thx.... BTW network:/// works when I manually type in file-manager

FYI, the scheme I'm referring to is the original scheme that gio/gvfs used. The file manager project wrapped the scheme and use its own way to deal with schemes so it might not match the schemes that you can passed to `gio`.

**Comment by Mile-Lile** [comment]
Also I founded [this](https://gitlab.gnome.org/GNOME/gvfs/-/issues/601)
so, multicast must be enabled...
